api/apps/api/test/scenario-run-results/world.ts

```diff
@@ -0,0 +1,114 @@
+import * as archiver from 'archiver';
+import * as unzipper from 'unzipper';
+import * as request from 'supertest';
+import { v4 } from 'uuid';
+import { createWriteStream, readFileSync, writeFileSync, unlinkSync } from 'fs';
+
+import { Repository } from 'typeorm';
+import { getRepositoryToken } from '@nestjs/typeorm';
+
+import { AppConfig } from '@marxan-api/utils/config.utils';
+import { MarxanExecutionMetadataGeoEntity } from '@marxan/marxan-output';
+import { DbConnections } from '@marxan-api/ormconfig.connections';
+
+import { bootstrapApplication } from '../utils/api-application';
+import { GivenUserIsLoggedIn } from '../steps/given-user-is-logged-in';
+import { GivenScenarioExists } from '../steps/given-scenario-exists';
+import { ScenariosTestUtils } from '../utils/scenarios.test.utils';
+import { GivenProjectExists } from '../steps/given-project';
+
+export const createWorld = async () => {
+  const app = await bootstrapApplication();
+  const token = await GivenUserIsLoggedIn(app);
+  const { projectId, cleanup } = await GivenProjectExists(app, token);
+  const scenario = await GivenScenarioExists(app, projectId, token);
+  const filesToRemove: string[] = [];
+
+  const marxanExecutionMetadataRepo: Repository<MarxanExecutionMetadataGeoEntity> = app.get(
+    getRepositoryToken(
+      MarxanExecutionMetadataGeoEntity,
+      DbConnections.geoprocessingDB,
+    ),
+  );
+
+  return {
+    cleanup: async () => {
+      filesToRemove.forEach((file) => unlinkSync(file));
+      await marxanExecutionMetadataRepo.delete({
+        scenarioId: scenario.id,
+      });
+      await ScenariosTestUtils.deleteScenario(app, token, scenario.id);
+      await cleanup();
+    },
+    GivenOutputZipIsAvailable: async () => {
+      const zipBuffer = await createArchive();
+      filesToRemove.push(...zipBuffer.filesToClean);
+      await marxanExecutionMetadataRepo.save(
+        marxanExecutionMetadataRepo.create({
+          outputZip: zipBuffer.archive,
+          inputZip: zipBuffer.archive,
+          scenarioId: scenario.id,
+        }),
+      );
+    },
+    WhenGettingZipArchive: async () =>
+      request(app.getHttpServer())
+        .get(`/api/v1/scenarios/${scenario.id}/marxan/output`)
+        .set('Authorization', `Bearer ${token}`),
+    ThenZipContainsOutputFiles: async (response: request.Response) => {
+      expect(response.header['content-type']).toEqual(
+        'application/octet-stream',
+      );
+      expect(response.header[`content-disposition`]).toEqual(
+        `attachment; filename="output.zip"`,
+      );
+      const directory = await unzipper.Open.buffer(response.body);
+      expect(directory.files.length).toEqual(1);
+      expect(directory.files[0].path).toEqual(`hello.txt`);
+      expect((await directory.files[0].buffer()).toString()).toEqual(
+        `Hello Marxan!`,
+      );
```

Comment:
wouldn't buffer comparison be enough?

Reply:
would need to save buffer in fixtures/world - just made a "shortcut" 😓

Reply:
ok :D